### ISSUE TITLE
Service Fabric New-AzureRmServiceFabricCluster generates a new certificate even when user passes -CertificateFile

### ISSUE THREAD

**Issue by JohnRGH** [Azure/service-fabric-issues#1022]
We generated a 2048 bit certificate using CertSetup.ps1 from the Service Fabric SDK (C:\Program Files\Microsoft SDKs\Service Fabric\ClusterSetup\Secure).  We then passed the certificate, template file, and parameter file, following the "Example 4" syntax here:
https://docs.microsoft.com/en-us/powershell/module/azurerm.servicefabric/new-azurermservicefabriccluster?view=azurermps-5.7.0

```
$pass="Password#1234" | ConvertTo-SecureString -AsPlainText -Force
$RGname="test20"
$keyVaultRG="test20kvrg"
$keyVault="test20kv"
$pfxsourcefile="c:\Mycertificates\my2017Prodcert.pfx"
$templateParmfile="~\Documents\GitHub\azure-quickstart-templates-parms\service-fabric-secure-nsg-cluster-65-node-3-nodetype\azuredeploytest.parameters.json"
$templateFile="~\GitHub\azure-quickstart-templates\service-fabric-secure-nsg-cluster-65-node-3-nodetype\azuredeploy.json"

New-AzureRmServiceFabricCluster -ResourceGroupName $RGname -TemplateFile $templateFile -ParameterFile $templateParmfile -CertificateFile $pfxsourcefile -CertificatePassword $pass -KeyVaultResouceGroupName $keyVaultRG -KeyVaultName $keyVault
```

First, I was prompted to also pass the -CertificateSubjectName, which is not shown as required in the documentation.  Then, the command created the cluster, but it generated a new certificate.  Upon downloading the certificate from the Azure portal, I found that it had "Intended Purposes" set to "Client Authentication" only (not "Server Authentication").  Due to this, I was unable to browse to Service Fabric Explorer after installing the certificate.

This took place last week in UK South region.  We have since had success in US East with New-AzureRmServiceFabricCluster and letting the command create its own certificate as output.

Please advise on the proper syntax for passing your own certificate, and the certificate creation behavior we observed.  Thank you.

**Comment by uditjh** [comment]
I'm trying Example 2 in order to utilize my own certificate from the vault for client to node security and getting the error message below 

https://docs.microsoft.com/en-us/powershell/module/azurerm.servicefabric/new-azurermservicefabriccluster?view=azurermps-5.7.0

PS C:\> New-AzureRmServiceFabricCluster -ResourceGroupName $RGname `
>> -TemplateFile $templateFile `
>> -ParameterFile $templateParmfile `
>> -SecretIdentifier $secretId
New-AzureRmServiceFabricCluster : Please specify the -certificateSubjectName
At line:1 char:1
+ New-AzureRmServiceFabricCluster -ResourceGroupName $RGname `
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzureRmServiceFabricCluster], PSInvalidOperationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ServiceFabric.Commands.NewAzureRmServiceFabricCluster


So on second attempt, I added the Certificate subject name along with a local copy of the same pfx file and then I get the error below.

PS C:\> New-AzureRmServiceFabricCluster -ResourceGroupName $RGname -TemplateFile $templateFile -ParameterFile $templateP
armfile -CertificateFile $pfxsourcefile -CertificatePassword $pass -CertificateSubjectName $certsubject -Verbose
New-AzureRmServiceFabricCluster : Parameter set cannot be resolved using the specified named parameters.
At line:1 char:1
+ New-AzureRmServiceFabricCluster -ResourceGroupName $RGname -TemplateF ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [New-AzureRmServiceFabricCluster], ParameterBindingException
    + FullyQualifiedErrorId : AmbiguousParameterSet,Microsoft.Azure.Commands.ServiceFabric.Commands.NewAzureRmServiceF
   abricCluster

My end goal is to able to utilize my own certificate to implement client to node security and not use the self signed certificate that is generated by NewAzureRM-ServiceFabricCluster command

The certificate was stored in the key vault as a secret (Base64 encoding for the certificate).

**Comment by juhacket** [comment]
@JohnRGH The "service authentication" issue has been fixed in Azure PowerShell 6.1.0.

@uditjh Please also try using the latest version of Azure PowerShell and see if it clears up the issues you are seeing. If you are still experiencing problems, please file a new issue for us to track.

**Comment by uditjh** [comment]
@juhacket I upgraded to AzureRM 6.6 and still cant follow the documentation listed [here](https://docs.microsoft.com/en-us/powershell/module/azurerm.servicefabric/new-azurermservicefabriccluster?view=azurermps-5.7.0#examples) Example 2 to resolve my issue of being able to use my own certificate when I standup service fabric cluster for client to node security.
 (Get-Module -ListAvailable | Where-Object{ $_.Name -eq 'AzureRM' }) ` | Select Version, Name, Author, PowerShellVersion | Format-List; 


Version           : 6.6.0
Name              : AzureRM
Author            : Microsoft Corporation
PowerShellVersion : 5.0
 New-AzureRmServiceFabricCluster -ResourceGroupName $rgName -TemplateFile $templatePath -ParameterFile $parametersPath -SecretIdentifier $secretId
New-AzureRmServiceFabricCluster : Please specify the -certificateSubjectName
At line:1 char:1
+ New-AzureRmServiceFabricCluster -ResourceGroupName $rgName -TemplateF ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzureRmServiceFabricCluster], PSInvalidOperationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ServiceFabric.Commands.NewAzureRmServiceFabricCluster
 

 New-AzureRmServiceFabricCluster -ResourceGroupName $rgName -TemplateFile $templatePath -ParameterFile $parametersPath -SecretIdentifier $secretId -CertificateSubjectName $certSubject
New-AzureRmServiceFabricCluster : Parameter set cannot be resolved using the specified named parameters.
At line:1 char:1
+ New-AzureRmServiceFabricCluster -ResourceGroupName $rgName -TemplateF ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [New-AzureRmServiceFabricCluster], ParameterBindingException
    + FullyQualifiedErrorId : AmbiguousParameterSet,Microsoft.Azure.Commands.ServiceFabric.Commands.NewAzureRmServiceFabricCluster
